docs/2a_general.md

```diff
@@ -6,129 +6,175 @@ The ThingSet protocol provides a consistent, standardized way to configure, moni
+All accessible data of a device is [structured as a tree](https://en.wikipedia.org/wiki/Tree_(data_structure)). The nodes of the tree structure are called **data objects** and correspond to the JSON object definition.
+
+Inner nodes in the structure are used to define the hierarchical structure of the data.
+
+Actual data is stored in the leaf nodes, called **data items**. The data items can contain any kind of measurements (e.g. temperature), device configuration (e.g. setpoint of a controller) or similar information.
+
+Each data object in the tree is identified by a numeric ID and a name. The ID can be chosen by the firmware developer. The name is a short case-sensitive ASCII string containing only alphanumeric characters, dots or underscores without any whitespace characters.
+
+The underscore is only used to specify the unit of a data item (also see below). No additional underscore is allowed in the name.
 
-Each node is identified by a unique node ID and a name. The ID can be chosen by the firmware developer. The name is a short case-sensitive ASCII string containing only alphanumeric characters, an underscore, dots or dashes without any whitespace characters. The underscore is only used to specify the unit of the data (if applicable, also see below). No additional underscore is allowed in the name and it should be unique per device at least for nodes that are used in publication messages.
+A dot is used to identify paths which are used internally by the implementation of the protocol itself (e.g. configuration of publication messages). Other usages of a dot in the data object names is not allowed.
 
-The numeric IDs are used to define the relations between nodes and to directly access data nodes in the binary protocol for reduced message size. For all interactions with users and in the text-based mode, only the node names and paths (consisting of multiple node names) are used.
+The IDs must be unique per device. Except for internal data objects (behind in a path starting with a dot) also the name must be unique.
+
+The IDs are used to access data objects in the binary protocol mode for reduced message size. They can also be used in the firmware to define the relations in the data structure. For all interactions with users and in the text-based mode, only the object names and paths are used.
+
+### Reserved IDs
+
+The IDs 0x10-0x1F are reserved for special data objects that need to be known in advance. In addition to that, IDs starting from 0x8000 are reserved for control purposes and will be assigned in the future.
+
+The following table shows the assigned IDs. Currently unassigned IDs might be defined in a future version of the protocol.
+
+| ID   | Name       | Description |
+|------|------------|-------------|
+| 0x10 | Time_s     | Unix timestamp in seconds since Jan 01, 1970 |
+| 0x17 | .name      | Endpoint used by binary protocol to determine names from IDs |
```

Comment:
Endpoint -> Data item?

Reply:
An endpoint is a specific type of data object / path which is used for a request (like in RESTful APIs). It's slightly different from a generic data object name, so I'd like to keep this here.